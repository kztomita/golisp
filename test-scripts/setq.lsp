(print (setq a 1))
(print (setq b 2 c (+ 1 2)))
(print (list a b c))

(print "====")
(print (setq outer_a 1))

(defun foo (a b)
  (let ((x 100) (y 200))
    (setq outer_a 100)	; overwrite
    (setq outer_b 200)	; new
    (setq b 20)		; overwrite parameter
    (setq y 30)		; overwite lexical variable
    (print (list outer_a outer_b a b x y))))


(print outer_a)
(foo 1 2)
(print (list outer_a outer_b))

