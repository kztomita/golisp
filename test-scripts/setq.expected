
1 
3 
(1 2 3) 
"====" 
1 
1 
(100 200 1 20 100 30) 
(100 200) 
